app/scripts/router.js

```diff
@@ -139,7 +141,7 @@ function (
       // back is enabled after the first view is rendered or
       // if the user is re-starts the app.
       this.canGoBack = this.window.sessionStorage.canGoBack || false;
-
+      this.environment = new Environment(this.window);
```

Comment:
Make this a pseudo-private variable by adding the `_` prefix. Allow an environment to be passed in for testing like:

```js
this.environment = options.environment || new Environment(this.window);
```

That way, we can pass in an environment for unit tests without reaching in to private variables like `this._environment` directly.